File: iotOnGolang/internal/model/DigitalTwin.go
package model

import (
    "time"
    "gorm.io/gorm"
)

//==============================================================
// DigitalTwin Declaration
//==============================================================
type DigitalTwin struct {
    gorm.Model
     TwinId                                    string
    DesiredStateVersion                                                            string
    ReportedStateVersion                                                            string
    LastSyncAt                                                            time.Time
    DeviceId         *uint
    Device           *IoTDevice `gorm:"foreignKey:DeviceId"`
    GatewayId         *uint
    Gateway           *Gateway `gorm:"foreignKey:GatewayId"`
    TemplateId         *uint
    Template           *TwinTemplate `gorm:"foreignKey:TemplateId"`
     ChangeEvents           []TwinChangeEvent `gorm:"foreignKey:ChangeEventsFromDigitalTwinId"`

// parent associations as their child

}

File: iotonrails/app/models/digitaltwin.rb

class DigitalTwin < ApplicationRecord


  has_many :Device, class_name: 'IoTDevice'
  has_many :Gateway, class_name: 'Gateway'
  has_many :Template, class_name: 'TwinTemplate'
  has_many :ChangeEvents, class_name: 'TwinChangeEvent'

end
